### PR TITLE
GH-11530 - Go major and minor version validation.

### PR DESCRIPTION
**Pull request by Hyaxia**
Go validation for a minimum minor version support.
When needing to up the minor version, just change the variable `MINIMUM_SUPPORTED_MINOR_VERSION` to the minimum version of golang that is needed.

Fixes https://github.com/mattermost/mattermost-server/issues/11530

### CODE REVIEW

**Comment by Hyaxia** [comment]
Please wait with merging this one, I have decided that I should add major and patch validations too, else it won't be specific enough.

**Comment by lieut-data** [comment]
@Hyaxia, adding major sounds good, but I'd probably skip patch: I don't think it would be /required/ to build as such.

**Comment by Hyaxia** [comment]
@lieut-data Okay I hear you, so I have added major check, ready to be reviewed and merged.

**Comment by lieut-data** [comment]
Thanks, @Hyaxia! I've asked @mkraft and @devinbinnie to take a look. Also, could you merge `origin/master` into your branch to get the build passing?

**Comment by Hyaxia** [comment]
@mkraft Fixed the way the validation is performed, please let me know if there are still problems, I have tested it again by myself, but I might have missed something.

**Comment by Hyaxia** [comment]
How do you suggest to clean the commits, I understand you don't want any force pushes.

**Comment by lieut-data** [comment]
@Hyaxia, no need to "clean" the commits -- we squash on merge, so that won't go into the history.

**Comment by mkraft** [comment]
I think it would be more in line with the conventions of this Makefile to have a target (ex `validate-go-version`) which can be used as a prerequisite to other targets, for example:

```diff
diff --git a/Makefile b/Makefile
index 60ed169d4..58e65aa77 100644
--- a/Makefile
+++ b/Makefile
@@ -454,7 +454,7 @@ test-data: start-docker ## Add test data to the local instance.
 	@echo Login with a regular account username=user-1 password=user-1
 	@echo ========================================================================
 
-define validate_go_version
+validate-go-version: ## Validates the installed version of go against Mattermost's minimum requirement.
 	@if [ $(GO_MAJOR_VERSION) -gt $(MINIMUM_SUPPORTED_GO_MAJOR_VERSION) ]; then \
 		exit 0 ;\
 	elif [ $(GO_MAJOR_VERSION) -lt $(MINIMUM_SUPPORTED_GO_MAJOR_VERSION) ]; then \
@@ -464,13 +464,10 @@ define validate_go_version
 		echo '$(GO_VERSION_VALIDATION_ERR_MSG)';\
 		exit 1; \
 	fi
-endef
 
-run-server: start-docker ## Starts the server.
+run-server: validate-go-version start-docker ## Starts the server.
 	@echo Running mattermost for development
 
-	$(call validate_go_version)
-
 	mkdir -p $(BUILD_WEBAPP_DIR)/dist/files
 	$(GO) run $(GOFLAGS) -ldflags '$(LDFLAGS)' $(PLATFORM_FILES) --disableconfigwatch | \
 	    $(GO) run $(GOFLAGS) -ldflags '$(LDFLAGS)' $(PLATFORM_FILES) logs --logrus &
```

**Comment by Hyaxia** [comment]
@mkraft absolutely agree, will perform the changes as you suggested, thanks!

File: Makefile [hunk 1]
```diff
@@ -449,9 +454,22 @@ test-data: start-docker ## Add test data to the local instance.
...
+		exit 1; \
+	elif [ $(GO_MINOR_VERSION) -lt $(MINIMUM_SUPPORTED_GO_MINOR_VERSION) ] ; then \
+		echo '$(GO_VERSION_VALIDATION_ERR_MSG)';\
+		exit 1; \
+	fi
+
+run-server: validate-go-version start-docker ## Starts the server.
 	@echo Running mattermost for development
 
+	$(call validate_go_version)
```
Comment (mkraft) [inline Makefile]: You can remove this call now that you're invoking the `validate-go-version` target as a prerequisite to `run-server`.

Comment (Hyaxia) [reply Makefile]: My bad, forgot to remove that one line, really late at night...

**Pull request merged**